src/app/home/home.component.ts

```diff
@@ -282,18 +317,29 @@ export class HomeComponent implements AfterViewInit {
         selection.addRange(range);
     }
 
+    /**
+     * Returns whether there is text in the editor or not.
+     */
     editorHasText(): boolean {
         return document.getElementById(this.EDITOR_KEY)!.innerHTML !== this.LINE_BROKEN_PARAGRAPH;
     }
 
+    /**
+     * Clear the written text in the editor.
+     */
     clearEditor(): void {
         document.getElementById(this.EDITOR_KEY)!.innerHTML = this.LINE_BROKEN_PARAGRAPH;
         this.processedText = undefined;
         this.updateCharacterAndWordCount();
         this.shouldCollapseSuggestions = new Array<boolean>(0);
     }
 
-    oscillateSuggestion(textMarkingIndex: number, $event: any) {
+    /**
+     * Expand or contract the suggestions of a given TextMarking based on an index.
+     * @param textMarkingIndex
+     * @param $event
```

Comment:
> Expand or contract the suggestions of a given TextMarking based on an index.
I couldn't understand this one from the comment alone. what does **based on index** mean 🤔 . What range of index makes it expand and which makes it contract. 

these two parameters need explanation here as this function is long enough to deserve that. 
- textMarkingIndex , specify maybe how the marking is indexed, does it start from 0 as expected or some other special rule, I wouldn't want to make assumptions on how this works, but rather have it in the comment
- $event - maybe add to which **element** this event is attached to on the html side, or I would have to guess.

Reply:
Addressed this to an extent, let me know if you'd like to change something about it